src/faebryk/exporters/visualize/interactive_graph.py

```diff
@@ -0,0 +1,214 @@
+# This file is part of the faebryk project
+# SPDX-License-Identifier: MIT
+
+import rich
+import rich.text
+
+from faebryk.core.core import GraphInterface, Link, Node
+from faebryk.core.graph import Graph
+from faebryk.core.util import get_all_connected, get_all_nodes_graph
+from faebryk.exporters.visualize.util import IDSet, generate_pastel_palette
+
+
+def interactive_graph(G: Graph):
+    import dash_cytoscape as cyto
+    from dash import Dash, html
+
+    # Register the fcose layout
+    cyto.load_extra_layouts()
+
+    app = Dash(__name__)
+
+    node_types: set[str] = set()
+    groups = {}
+
+    def _group(gif: GraphInterface) -> str:
+        node = gif.node
+        my_node_id = str(id(node))
+        if my_node_id not in groups:
+            label = f"{node.get_full_name()} ({type(node).__name__})"
+            groups[my_node_id] = {
+                "data": {
+                    "id": my_node_id,
+                    "label": label,
+                    "type": "group",
+                }
+            }
+
+        return my_node_id
+
+    def _node(node: Node):
+        full_name = node.get_full_name()
+        type_name = type(node).__name__
+        node_types.add(type_name)
+        data = {"id": str(id(node)), "label": full_name, "type": type_name}
+        if isinstance(node, GraphInterface):
+            data["parent"] = _group(node)
+        return {"data": data}
+
+    link_types: set[str] = set()
+    links_touched = IDSet[Link]()
+
+    def _link(link: Link):
+        if link in links_touched:
+            return None
+        links_touched.add(link)
+
+        try:
+            source, target = tuple(str(id(n)) for n in link.get_connections())
+        except ValueError:
+            return None
+
+        type_name = type(link).__name__
+        link_types.add(type_name)
+
+        return {"data": {"source": source, "target": target, "type": type_name}}
+
+    all_gifs = list(
+        gif for node in get_all_nodes_graph(G) for gif in node.GIFs.get_all()
```

Comment:
you can get the gifs with node.GIFs.self.edges instead of get_all to make it graph space

Reply:
but all_gifs = G.nodes

Reply:
Ah nice! Thanks mate! Much cleaner!